test/e2e/framework/util.go

```diff
@@ -5192,12 +5192,13 @@ func GetClusterZones(c clientset.Interface) (sets.String, error) {
 	if err != nil {
 		return nil, fmt.Errorf("Error getting nodes while attempting to list cluster zones: %v", err)
 	}
-
 	// collect values of zone label from all nodes
 	zones := sets.NewString()
 	for _, node := range nodes.Items {
 		if zone, found := node.Labels[kubeletapis.LabelZoneFailureDomain]; found {
 			zones.Insert(zone)
+		} else {
+			return zones, fmt.Errorf("zone name for node %s not found. No label with key %s", node.Name, kubeletapis.LabelZoneFailureDomain)
```

Comment:
No current caller of this method appears to need this constraint.  Are you sure the zone test actually needs it?  It's not what I would expect.

Reply:
I see the original getZoneNames() fails as soon as a node is missing zone label. So I add this constraint.

Reply:
> I see the original getZoneNames() fails as soon as a node is missing zone label. So I add this constraint.

But that is one of two functions you're combining. They have different semantics and after your pull the semantics that drove the original will now be wrong.  Either constraints of `getZoneNames` needs to be relaxed (I don't know if that is legal) or you cannot combine this.

Reply:
OK, Thanks for your review.

Reply:
IMO this is OK. When I wrote the other call to `GetClusterZones()`, it didn't expect the node label to not exist but it should have, so I see this as a bonus.

Reply:
> IMO this is OK. When I wrote the other call to GetClusterZones(), it didn't expect the node label to not exist but it should have, so I see this as a bonus.

If included in a conformance tests (and our frameworks might be) it is overly restrictive on topology in a way that conformance shouldn't be.